src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -34,136 +33,125 @@ namespace System
           IMinMaxValue<nint>,
           ISignedNumber<nint>
     {
-        private readonly unsafe void* _value; // Do not rename (binary serialization)
```

Comment:
(nit) Do we want to keep the comment here, since we still support binary serialization (for now).

Reply:
@GrabYourPitchforks indicated the comment is out of date since we have an explicit serializer method.

Reply:
To expand on this a little bit, the _ISerializable_ interface tells reflection-based serializers, "Stop! There's something special about my layout and you shouldn't perform your normal scrape-the-fields logic when serializing me. Instead, call _GetObjectData_ and my special ctor, and I'll tell you how to serialize my own instances."

So as long as the type has _ISerializable_ on it (or we add it), we have considerable leeway to change the internal layout of the type with regard to serialization. Debuggers (SOS) may need to be updated separately, though.